modules/services/git-workspace.nix

```
@@ -0,0 +1,153 @@
+      default = "daily";
+      description = ''
+        The refresh frequency. Check <literal>man systemd.time</literal> for
+        more information on the syntax.
+      '';
+    };
+    environmentFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        Path to the systemd <literal>EnvironmentFile</literal> to keep tokens,
+        do not put a derivation output here since it would expose secrets in
+        the world readable Nix store.
+            </para><para>
+        It contains lines like
+        <literal>
+        GITHUB_TOKEN=ghp_...
+        </literal>
+            </para><para>
+        If you want to use Nix to manage these tokens use a tool like sops-nix
+        or agenix. This file must be readable by the home-manager user.
+      '';
+      example = "/var/lib/git-workspace-tokens";
+    };
+    workspaces = lib.mkOption {
+      type = lib.types.attrsOf workspaceType;
+      default = { };
+      description = ''
+        1:1 representation of a single <literal>workspace.toml</literal> consumed
+        by git-workspace.
+      '';
+      example = lib.literalExpression ''
+        {
+          personal.provider = [
+            {
+              provider = "github"
+              name = "github_username"
+              path = "''${config.home.homeDirectory}/personal_projects"
+              skip_forks = false;
+            }
+            {
+              provider = "gitlab"
+              name = "gitlab_username"
+              # it's possible to use the same path as previous provider
+              # if there are not name clashes
+              path = "''${config.home.homeDirectory}/personal_projects"
+            }
+          ];
+          work.provider = [
+            {
+              provider = "github"
+              name = "github_org"
+              path = "''${config.home.homeDirectory}/work"
+            }
+          ];
+        };
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      (lib.hm.assertions.assertPlatform "services.git-workspace" pkgs
+        lib.platforms.linux)
+    ];
+    xdg.configFile = lib.mapAttrs' (workspaceName: workspace: {
+      name = "git-workspace/${workspaceName}/workspace.toml";
+      value.source = (tomlFormat.generate "${workspaceName}-workspace.toml"
+        workspace).outPath;
+    }) cfg.workspaces;
+    systemd.user.services = lib.mapAttrs' (workspaceName: workspace: rec {
+      name = "git-workspace-${workspaceName}-update";
+      value = {
+        Unit.Description = "Runs `git-workspace update` for ${workspaceName}";
+        Service = {
+          EnvironmentFile = cfg.environmentFile;
+          ExecStart = let
+            script = pkgs.writeShellApplication {
+              name = "${name}-launcher";
+              text = ''
+                ${cfg.package}/bin/git-workspace \
+                  --workspace ${config.xdg.configHome}/git-workspace/${workspaceName} \
+                  update
```

I think this may have problems if `configHome` or `workspaceName` contains, e.g., whitespaces. Perhaps can do

```suggestion
                  ${escapeShellArgs [ "--workspace" "${config.xdg.configHome}/git-workspace/${workspaceName}"
                  "update"]}
```